pkg/extension/sumologicextension/extension.go

```diff
@@ -137,43 +128,46 @@ func (se *SumologicExtension) checkCollectorCredentials() bool {
 }
 
 // getCollectorCredentials retrieves and decrypts collector credentials using
-// collector name as passphrase.
-func (se *SumologicExtension) getCollectorCredentials() error {
-	filenameHash, err := createHash(se.conf.CollectorName)
+// hashed collector name as passphrase.
+func (se *SumologicExtension) getCollectorCredentials() (OpenRegisterResponsePayload, error) {
```

Comment:
What is the value added of returning `OpenRegisterResponsePayload` and assign it to `se` struct outside the function instead of make it in place in this method?

Reply:
This IMHO makes it obvious what the function does: it's called `get...` so it gets it hence the return value. Setting this field in the function is a non obvious side effect.